src/main/kotlin/dev/robustum/core/recipe/RobustumRecipeSerializers.kt

```diff
@@ -0,0 +1,136 @@
+package dev.robustum.core.recipe
+
+import com.mojang.serialization.Codec
+import com.mojang.serialization.codecs.RecordCodecBuilder
+import dev.robustum.core.RobustumCore
+import dev.robustum.core.codec.RobustumCodecs
+import dev.robustum.core.extensions.toDefaultedList
+import net.minecraft.item.ItemStack
+import net.minecraft.recipe.*
+import net.minecraft.util.Identifier
+import net.minecraft.util.registry.Registry
+
+object RobustumRecipeSerializers {
+    @JvmField
+    val SHAPELESS: DelegatedRecipeSerializer<ShapelessRecipe> =
+        registerDelegated(
+            "shapeless",
+            RecipeSerializer.SHAPELESS,
+        ) { id: Identifier ->
+            RecordCodecBuilder.mapCodec<ShapelessRecipe> { instance ->
+                instance
+                    .group(
+                        Codec.STRING
+                            .optionalFieldOf("group", "")
+                            .forGetter(ShapelessRecipe::getGroup),
+                        RobustumCodecs.NON_EMPTY_INGREDIENT
+                            .listOf()
+                            .fieldOf("inputs")
+                            .forGetter(ShapelessRecipe::getIngredients),
+                        RobustumCodecs.ITEM_STACK
+                            .fieldOf("output")
+                            .forGetter(ShapelessRecipe::getOutput),
+                    ).apply(instance) { group: String, ingredients: List<Ingredient>, result: ItemStack ->
+                        ShapelessRecipe(id, group, result, ingredients.toDefaultedList(Ingredient.EMPTY))
+                    }
+            }
+        }
+
+    @JvmField
+    val SMELTING: DelegatedRecipeSerializer<SmeltingRecipe> =
```

Comment:
```suggestion

    /**
     * @sample test.recipe.TestRobustumRecipes.testRecipe
     */
    @JvmField
    val SMELTING: DelegatedRecipeSerializer<SmeltingRecipe> =
```

せっかくテスト書きましたしこんな感じでサンプルとして載せておくのはどうでしょう？
IDEAからはこんな感じに見えます
![image](https://github.com/user-attachments/assets/30cf5c46-9f39-4440-82b3-ca8cce308591)

Reply:
めちゃくちゃありですね

Reply:
これやってみたんすけど，`TestRobustumRecipes`を参照できてません

Reply:
上で書いた通りにして駄目でした？普通できるはずなんですけど...